pipeline/stop_places/sql/stop_places.sql

```diff
@@ -127,6 +203,199 @@ $$
+    );
+  -- ramp/slope:
+  ELSEIF tags->>'highway' = 'footway' AND tags->>'incline' IS NOT NULL
+    THEN result := xmlconcat(result,
+      create_KeyValue('2120', ''::text),  -- ramp or slope
+      --delfi_attribute_get_value_xml('2122', tags->>),  -- ramp length
+      delfi_attribute_get_value_xml('2123', tags->>'width'),  -- ramp width
+      -- TODO: convert slope to degrees
+      delfi_attribute_get_value_xml('2124', tags->>'incline')  -- ramp slope
+    );
+  END IF;
+
+  IF result IS NOT NULL THEN
+    RETURN xmlelement(name "keyList", result);
+  END IF;
+
+  RETURN NULL;
+END
+$$
+LANGUAGE plpgsql IMMUTABLE;
+
+
+/*
+ * Create a keyList element for access spaces based on a delfi attribut to osm matching
+ * Optionally additional key value pairs can be passed to the function
+ * Returns null when no tag matching exists
+ */
+CREATE OR REPLACE FUNCTION ex_keyList_AccessSpace(tags jsonb, additionalPairs xml DEFAULT NULL) RETURNS xml AS
+$$
+DECLARE
+  result xml;
+BEGIN
+  result := additionalPairs;
+
+  -- barrier/blocking element/narrow section
+  IF tags->>'barrier' = 'cycle_barrier' OR tags->>'crossing:chicane' = 'yes'
+    THEN result := xmlconcat(result, create_KeyValue('2080', ''::text));  -- bicycle barrier
+  END IF;
+  -- delfi_attribute_on_yes_xml('2081', tags->>)  -- movement area into, through and out of the narrow section
+
+  -- elevator:
+  -- just the entry (door) of the elevator is considered as an access space
+  IF tags->>'door' IS NOT NULL
+    THEN result := (result, delfi_attribute_get_value_xml('2091', tags->>'width'));  -- door width of the lift
+  END IF;
+
+  IF result IS NOT NULL THEN
+    RETURN xmlelement(name "keyList", result);
+  END IF;
+
+  RETURN NULL;
+END
+$$
+LANGUAGE plpgsql IMMUTABLE;
+
+
+/*
+ * Create a keyList element for entrances based on a delfi attribut to osm matching
+ * Optionally additional key value pairs can be passed to the function
+ * Returns null when no tag matching exists
+ */
+CREATE OR REPLACE FUNCTION ex_keyList_Entrance(tags jsonb, additionalPairs xml DEFAULT NULL) RETURNS xml AS
+$$
+DECLARE
+  result xml;
+BEGIN
+  result := xmlconcat(
+    additionalPairs,
+    delfi_attribute_on_yes_xml('2030', tags->>'entrance'),  -- entrance
+    delfi_attribute_on_yes_xml('2031', tags->>'opening_hours')  -- opening hours
+  );
+
+  -- type of entrance:
+  IF tags->>'door' = 'yes'
+    THEN result := xmlconcat(result, delfi_attribute_on_yes_xml('2032', 'Tür'));
+  ELSEIF tags->>'door' = 'hinged'
+    THEN result := xmlconcat(result, delfi_attribute_on_yes_xml('2032', 'Anschlagtür'));
+  ELSEIF tags->>'door' = 'sliding'
+    THEN result := xmlconcat(result, delfi_attribute_on_yes_xml('2032', 'Schiebetür'));
+  ELSEIF tags->>'door' = 'revolving'
+    THEN result := xmlconcat(result, delfi_attribute_on_yes_xml('2032', 'Drehtür'));
+  ELSEIF tags->>'door' = 'swinging'
+    THEN result := xmlconcat(result, delfi_attribute_on_yes_xml('2032', 'Pendeltür'));
+  END IF;
```

Comment:
I would prefer if possible to keep the concat method around all attribute selections (see also comment regarding `create_keyList`)
Also to shorten the code and improve readability the if/elseif checks can be nested in the `delfi_attribute_..._xml` function (code is untested, maybe extra `()` or so are required).

```SQL
xmlconcat(
   delfi_attribute_on_yes_xml('2032',
    SELECT CASE
      WHEN $1 = 'yes' THEN 'Tür'
      WHEN $1 = 'hinged' THEN 'Anschlagtür'
    END
   )
);
```


I also think that there is a mistake here.
`delfi_attribute_on_yes_xml('2032', 'Schiebetür'));` (and all the others) will always return `NULL` since the function checks whether the given value equals "yes" which is never the case.